mapper/software_management/src/message.rs

```diff
@@ -0,0 +1,110 @@
+use crate::software::*;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Deserialize, Serialize, PartialEq)]
+pub struct SoftwareRequest {
+    pub id: String,
```

Comment:
This `id` must be optional as Cumulocity doesn't send any id in the software update SmartREST operation messages. The sm-agent will have to generate and maintain these ids for request-response tracking.